### PR TITLE
More timer fixups

### PR DESCRIPTION
**Pull request by b-man**
Insert ISBs after writes to the control register and before reads
from the counter register to ensure that access to these registers
always occur in an orderly manner.

To paraphrase ARM DDI 0487J.a D11.2.1/D11.2.2:
Reads from the counter register can occur speculatvely and out of order
relative to other instructions executed on the PE.

To quote ARM DDI 0487J.a Glossary - Context Synchronization event:
"All direct and indirect writes to System registers that are made before the Context synchronization event
affect any instruction, including a direct read, that appears in program order after the instruction causing the
Context synchronization event."

Since the cyclic subsystem serves as a time source for the
rest of the kernel it really should have full control over
the timers on all CPUs. Allow cbe_enable/disable to set or
clear the ENABLE bit for the timer, rather than just the IMASK bit.

### CODE REVIEW

File: usr/src/psm/stand/boot/aarch64/common/arch_timer.c [hunk 1]
```diff
@@ -63,6 +63,7 @@ arch_timer_mask_irq(void)
 	ctl = ops->read_cnt_ctl();
 	ctl |= CNT_CTL_IMASK;
 	ops->write_cnt_ctl(ctl);
+	isb();
```
Comment (richlowe) [inline usr/src/psm/stand/boot/aarch64/common/arch_timer.c]: Would it make sense to put the `isb` in the write/read implementations, and not after each call to them?  That way they can't be forgotten, and we can leave a comment saying why more easily.

Comment (b-man) [reply usr/src/psm/stand/boot/aarch64/common/arch_timer.c]: I can do that, however if we do that then should we also apply this pattern to all of the other applicable registers in controlregs.h?

There are several other examples of controlreg calls immediately followed by an isb() (write_s1e1r(), write_sctlr(), write_cpacr_el1(), write_ttbr0(),  write_tcr(), and write_vbar()).

The isb() preceding read() in this specific case is more to account for synchronization in tight loops (a plausible scenario for arch_timer_count()).

Comment (richlowe) [reply usr/src/psm/stand/boot/aarch64/common/arch_timer.c]: It feels like if we need the barrier all the time, we should put the barrier in the function that does the read, yeah, but I'm ok with you not doing that right now for all the other registers, and we can do that separately.

Comment (b-man) [reply usr/src/psm/stand/boot/aarch64/common/arch_timer.c]: Sounds good. I'll update this later today with your recommended change.

**Comment by richlowe** [comment]
I'm confused by what's happened here now?  What did I miss?

**Comment by b-man** [comment]
> I'm confused by what's happened here now? What did I miss?

Sorry about that - seems I accidentally closed the PR after resetting my branch. Everything should be all set now.

**Review by richlowe — APPROVED** [review]

**Comment by richlowe** [comment]
merged

**Pull request closed without merging**